EIPS/eip-7923.md

```diff
@@ -145,21 +147,19 @@ index bf34fbee..477f969e 100644
 +        start_page = start_position >> 12
 +        end_page = end >> 12
 +
-+        gas = 0
-+
 +        for page in range(start_page, end_page + 1):
 +            if page not in self._memory.pages:
-+                gas += ALLOCATE_PAGE_COST
++                reason = f"Allocating page {page << 12}"
++                self._gas_meter.consume_gas(ALLOCATE_PAGE_COST, reason)
 +
 +            if page not in self._memory.lru_pages:
-+                gas += THRASH_PAGE_COST
-+
-+        for page in range(start_page, end_page + 1):
-+            self._memory.lru_pages[page] = True
-
--            self._memory.extend(start_position, size)
-+        reason = ""  # dummy reason
-+        self._gas_meter.consume_gas(gas, reason)
++                reason = f"Page {page << 12} not in LRU pages"
```

Comment:
Nit: make the `12` a constant somewhere. So then `PAGE_SIZE = 2^PAGE_BITS` or something

Reply:
07bee6723f615078688c43e9f3f794bf8b35c68a